Automattic/wp-calypso
# Advanced SEO: Add Reader Preview

Adds a basic Reader Preview to the Advanced SEO Social Preview area of the editor preview. It won't be a 100% accurate representation of what a user will see in the Reader, but it's a good start :)

It was decided for this iteration to use the same classNames and styles from the `reader/stream/post` component. In the near future we can switch this component to use a new Reader post card that is currently in development by the Reader squad.

It uses a few utilities that the Reader offers, which means I need to add the server side rendering flag to a few of the files. I also split out a `formatExcerpt` method in `lib/post-normalizer/rule-create-better-excerpt.js` so that we can get a nicely formatted post excerpt from the content.

**To Test**
* Ensure your site has a business plan.
* Create/edit a post and preview it in the Social sharing tab. You should see the Reader preview there.
* It should pick up an image to preview if it is large enough.
* Try it in multiple screen sizes!

<img width="931" alt="screen shot 2016-08-08 at 12 08 59 pm" src="https://cloud.githubusercontent.com/assets/789137/17492528/0263044a-5d61-11e6-8904-125a090352dd.png">


Test live: https://calypso.live/?branch=add/advanced-seo-reader-preview
